backend/app/api/v1/endpoints/company.py

```diff
@@ -26,3 +26,16 @@ async def get_all_companies(db_session=Depends(get_db)):
     """
     
     return CompanyRepository(db_session).get_all_companies()
+
+@router.get("/{id}", status_code=status.HTTP_200_OK, response_model=Company)
+async def get_company_by_id(id: str, db_session=Depends(get_db)):
+    """
```

Comment:
While block comments are good, and are a flavour of documentation. There is something called the clean code principle. If you name your variables and functions correctly. Code speaks for itself. IF its a complex algo, large return block from call, or some data dump response, then you do block comments.